tests/unit/bootstrap.php

```diff
@@ -98,6 +98,13 @@ function() {
 	require_once $_plugin_dir . 'includes/admin/class-wc-rest-payments-payment-intents-controller.php';
 	require_once $_plugin_dir . 'includes/class-woopay-tracker.php';
 	require_once $_plugin_dir . 'includes/admin/class-wc-rest-payments-customer-controller.php';
+
+	DG\BypassFinals::enable( false, true );
+	DG\BypassFinals::setWhitelist(
+		[
+			'*/AbstractSessionRateLimiter.php',
+		]
+	);
```

Comment:
@RadoslavGeorgiev - what do you think of introducing this package to overcome the limitation of PHPUnit and `final` keywords? 

> Please note that final, private, and static methods cannot be stubbed or mocked. 
> -- Source: https://docs.phpunit.de/en/9.6/test-doubles.html

FYI. Woo core is literally using the same approach and package https://github.com/woocommerce/woocommerce/blob/70154d5476254a061734c0bf47259d9a71156aca/plugins/woocommerce/tests/Tools/CodeHacking/CodeHacker.php#L18-L19

Reply:
Another approach is to extract an interface for `AbstractSessionRateLimiter` and we're going to provide this interface in `PaymentsServiceProvider` rather than the concrete class `FailedTransactionRateLimiter`.

Reply:
The interfact approach is possible, however the downside is that every time we add new methods into `AbstractSessionRateLimiter`, we also need to add it to the interface. It's not a specific issue here as I don't oversee that we will frequently add more methods into  `AbstractSessionRateLimiter`. Nonetheless, that can happen for other future classes using `final` keywords. 

In general, I prefer the approach of using the new package `DG\BypassFinals`. Otherwise, at least we can ask Woo core team to extract their CodeHacker to a PHP composer package so that we can re-use the code in WooPayments.

Reply:
The `dg/bypass-finals` package looks good to me, excellent job figuring this out!

We might also keep the `CodeHacker` class in mind for the future, at some point, something similar would be good for `src`.